tpl/tplimpl/template_funcs_test.go

```diff
@@ -1021,6 +1023,52 @@ func TestIntersect(t *testing.T) {
 	}
 }
 
+func TestUnion(t *testing.T) {
+	t.Parallel()
+	for i, this := range []struct {
+		sequence1 interface{}
+		sequence2 interface{}
+		expect    interface{}
+	}{
+		{[]string{"a", "b", "c", "c"}, []string{"a", "b", "b"}, []string{"a", "b", "c"}},
+		{[]string{"a", "b"}, []string{"a", "b", "c"}, []string{"a", "b", "c"}},
+		{[]string{"a", "b", "c"}, []string{"d", "e"}, []string{"a", "b", "c", "d", "e"}},
+		{[]string{}, []string{}, []string{}},
+		{[]string{"a", "b"}, nil, make([]interface{}, 0)},
+		{nil, []string{"a", "b"}, make([]interface{}, 0)},
+		{nil, nil, make([]interface{}, 0)},
+		{[]string{"1", "2"}, []int{1, 2}, make([]string, 0)},
```

Comment:
The [intersection func](https://github.com/spf13/hugo/blob/master/tpl/tplimpl/template_funcs.go#L352-L353) also enforces that both array have to be of the same type. `[]string{}` ≠ `[]int{}`.

If there is some ambiguity in play we could just return the first or second slice but I prefer to be more explicit about this error so that users aren't confused at the end.